Library/ViewModels/BackingViewModel.swift

```diff
@@ -238,45 +250,118 @@ public final class BackingViewModel: BackingViewModelType, BackingViewModelInput
   public let backerName: Signal<String, NoError>
-  public let backerNameAccessibilityLabel: Signal<String, NoError>
-  public let backerPledgeAmountAndDate: Signal<String, NoError>
-  public let backerPledgeAmountAndDateAccessibilityLabel: Signal<String, NoError>
-  public let backerPledgeStatus: Signal<String, NoError>
-  public let backerPledgeStatusAccessibilityLabel: Signal<String, NoError>
-  public let backerRewardDescription: Signal<String, NoError>
-  public let backerRewardDescriptionAccessibilityLabel: Signal<String, NoError>
   public let backerSequence: Signal<String, NoError>
-  public let backerSequenceAccessibilityLabel: Signal<String, NoError>
-  public let backerShippingAmount: Signal<String, NoError>
-  public let backerShippingAmountAccessibilityLabel: Signal<String, NoError>
-  public let backerShippingDescription: Signal<String, NoError>
-  public let backerShippingDescriptionAccessibilityLabel: Signal<String, NoError>
-  public let estimatedDeliveryDateLabelText: Signal<String, NoError>
-  public let estimatedDeliveryStackViewHidden: Signal<Bool, NoError>
   public let goToMessageCreator: Signal<(MessageSubject, Koala.MessageDialogContext), NoError>
   public let goToMessages: Signal<(Project, Backing), NoError>
-  public let hideActionsStackView: Signal<Bool, NoError>
-  public let loadingOverlayIsHidden: Signal<Bool, NoError>
+  public let loaderIsAnimating: Signal<Bool, NoError>
   public let messageButtonTitleText: Signal<String, NoError>
+  public let opacityForContainers: Signal<CGFloat, NoError>
+  public let pledgeAmount: Signal<String, NoError>
+  public let pledgeSectionTitle: Signal<NSAttributedString, NoError>
+  public let rewardDescription: Signal<String, NoError>
+  public let rewardSectionAndShippingIsHidden: Signal<Bool, NoError>
+  public var rewardTitleWithAmount: Signal<String, NoError>
+  public var rewardSectionTitle: Signal<NSAttributedString, NoError>
   public let rootStackViewAxis: Signal<UILayoutConstraintAxis, NoError>
+  public let shippingAmount: Signal<String, NoError>
+  public let statusDescription: Signal<NSAttributedString, NoError>
+  public let totalPledgeAmount: Signal<String, NoError>
 
   public var inputs: BackingViewModelInputs { return self }
   public var outputs: BackingViewModelOutputs { return self }
 }
 
-private func statusString(_ forStatus: Backing.Status) -> String {
-    switch forStatus {
-    case .canceled:
-      return Strings.project_view_pledge_status_canceled()
-    case .collected:
-      return Strings.project_view_pledge_status_collected()
-    case .dropped:
-      return Strings.project_view_pledge_status_dropped()
-    case .errored:
-      return Strings.project_view_pledge_status_errored()
-    case .pledged:
-      return Strings.project_view_pledge_status_pledged()
-    case .preauth:
-      fatalError()
+private func statusDescString(for backing: Backing, project: Project, backerIsCurrentUser: Bool)
+  -> NSAttributedString {
+
+  var string = ""
```

Comment:
I think if the `switch` is exhaustive this can be a `let string: String` and then it can be instantiated in the `switch` just for fun!